rllib/policy/eager_tf_policy_v2.py

```diff
@@ -178,7 +178,19 @@ def loss(
         Returns:
             A single loss tensor or a list of loss tensors.
         """
-        raise NotImplementedError
+        # Under the new _enable_learner_api the loss function still gets called in order
+        # to initialize the view requirements of the sample batches that are returned by
+        # the sampler. In this case, we don't actually want to compute any loss, however
+        # if we access the keys that are needed for a forward_train pass, then the
+        # sampler will include those keys in the sample batches it returns. This means
+        # that the correct sample batch keys will be available when using the learner
+        # group API.
+        if self.config._enable_learner_api:
+            for k in model.input_specs_train():
+                train_batch[k]
```

Comment:
what is this doing? `train_batch[k]`?

Reply:
this accesses the key of the train batch, so that train_batch.accessed keys includes k.

in this case the train_batch that is getting accessed is the dummy batch that is being used to initialize the view requirements.

Reply:
I'll add an additional comment to make it more clear what is happening here.

Reply:
so with this change, no-one needs to create any policy instance for any of the new algorithms? This is simplifying the migration guide?

Reply:
yes exactly